packages/client/src/Components/Mapping/helpers/submitClearLastLeg.js

```diff
@@ -1,10 +1,8 @@
 /** the user has clicked on the end of a route, but he just wants to
    * drop the last leg
    */
-const submitClearLastLeg = (/* dictionary */ plannedLegs, /* string */ turnName, /* object */ planningMarker) => {
-  // pop the last leg in the dictionary
-
-  // update the planning marker
+const submitClearLastLeg = (/* dictionary */ plannedLegs) => {
+  return plannedLegs.reduce((d, i, idx, l) => idx < l.length - 1 ? [...d, i] : d, [])
```

Comment:
oh dear, it will be a while before my JS skills are up to this standard.

Is there an 'old fashioned' version that's more accessible to this luddite?

Reply:
Done. It now looks like this:

```
const submitClearLastLeg = (/* dictionary */ plannedLegs) => {
  const newPlannedLegs = plannedLegs
  newPlannedLegs.pop()
  return newPlannedLegs
}
```